students/Lincoln_Zhang/Session4/mailroom_part2.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+import sys,os,textwrap
+
+print("Mailroom")
+
+donors_db = {"Fred Flintstone":[100, 200, 50],
+          "Barney Rubble":[100, 50, 600],
+          "Wilma Flintstone":[1000, 50, 600, 200],
+          "Pebbles Flintstone":[10, 5, 6]
+          }
+
+
+def Mail_to_donors(a_dict):
+    """
+    This Function is to generate Thank you letter to all donors in donors_db
+
+    and store the letters in letters.txt file
+    """
+    with open ("letters.txt","w+") as f:
+        for item in a_dict.items():
+            name,amount = item
+            amount = sum(amount)
+            letter = textwrap.dedent("""
+            Dear {},
+
+            Thank you! The amount you donated is {}!
+
+            Sincerely
+            The Team
+
+            """.format(name,amount))
+            print(letter)
+            f.write(letter)
+
+def find_donor(name,cash):
+    """
+    This Function is to verify db with donor's name
+    if this is the donor first donation, create a new record in donors_db
+    otherwise, append this donation to donor's record list
+    """
+    if name in donors_db.keys():
+        donors_db[name] = donors_db[name].append(cash)
+    else:
+        donation = []
+        donor = {name:donation}
+        donor[name].append(cash)
+        donors_db.update(donor)
+    return donor
+
+def Thank_You():
+    """
+    This function is to generate thank you letter for each donation
+    and store the letter by donor's name
+    """
+    donor_name = input("Please tell me your name: ")
+    donation = input("Please tell me how much you would like to donate: ")
+    find_donor(name = donor_name, cash = int(donation))
+    print(donors_db)
+    with open("{}.txt".format(donor_name),"w+") as f:
```

Comment:
notice that this letter text is the same as above? That's why you want it in its own function.